kartothek/core/dataset.py

```diff
@@ -408,6 +415,7 @@ def get_indices_as_dataframe(
 
         # start joining with the small ones
         sorted_dfs = sorted(dfs, key=lambda df: len(df))
+        # TODO: Why not do a pd.concat here?
```

Comment:
probably faster and easier to read, you are right. Not sure why we haven't used this, probably because the one who implemented it (which I believe was me) didn't think of it at the time.

Unfortunately we don't have any ready-to-use benchmarks for this function yet. That would be a nice thing to benchmark. If you feel like it, you can add a case to `asv_bench/benchmarks` and give it a shot.

cc @NeroCorleone I believe this is the place where the memory spike you're investigating occurs. You might want to check out the concat instead of the merge as well for the high mem scenarios

I guess, this would be a good change for a followup PR, though, to not mix up too much stuff in one go.